flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSink.java

```diff
@@ -227,11 +226,13 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
                                         keyProjection,
                                         flushMode,
                                         objectReuse
-                                                ? createRowDataTypeSerializer(
-                                                                context,
-                                                                dataStream.getExecutionConfig())
-                                                        ::copy
-                                                : Function.identity());
+                                                ? (rowData) ->
+                                                        createRowDataTypeSerializer(
+                                                                        context,
+                                                                        dataStream
+                                                                                .getExecutionConfig())
+                                                                .copy(rowData)
```

Comment:
Afaik a method lambda should also work (if not I'm happy to learn that ;)), so you wouldn't need to change that part.

Reply:
Works but Intellij shows the error `Cannot infer arguments` :)